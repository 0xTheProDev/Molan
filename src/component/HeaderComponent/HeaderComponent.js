// @flow
import React from 'react';
import { Image } from 'semantic-ui-react';
import Clock from 'react-live-clock';

import Logo from 'assets/molan_logo.png';

import Styles from './HeaderComponent.module.css';

export function HeaderComponent() {
  return (
    <div className={Styles['header-content']}>
      <div className={Styles['brand']} role="banner">
        <Image src={Logo} size="small" alt="Molan iDE" />
      </div>
      <div className={Styles['clock-area']}>
        <i aria-hidden="true" className="clock outline icon" />
        <Clock ticking format={"hh:mm:ssa"} />
      </div>
    </div>
  );
}
