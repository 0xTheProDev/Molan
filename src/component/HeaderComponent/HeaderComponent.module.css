.header-content {
  background-color: #FFF !important;
  height: 65px !important;
}

.brand {
  padding-left: 40px !important;
  margin-top: -20px !important;
  user-select: none !important;
  pointer-events: none !important;
  -webkit-user-drag: none !important;
}

.clock-area {
  position: absolute !important;
  margin-left: auto !important;
  right: 10px !important;
  top: 5px !important;
  cursor: pointer;
  font-size: 16px;
  padding: 5px 10px;
  border: 1px solid rgba(0,0,0,.4);
  border-radius: 10px;
  transition: all .5s ease;
}

.clock-area:hover {
  box-shadow: 1px 1px rgba(0,0,0,.6);
}
