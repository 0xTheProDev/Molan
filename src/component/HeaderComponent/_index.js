// @flow
export { HeaderComponent, HeaderComponent as default } from './HeaderComponent';
