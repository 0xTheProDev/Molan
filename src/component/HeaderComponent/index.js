import React, { Component } from 'react';
import { Image } from 'semantic-ui-react';
import logo from './molan.png';
import './index.css';

export default class HeaderComponent extends Component {
    render() {
        return (
            <div className='header-content'>
                <div className='brand'>
                    <Image src={logo} size='small'/>
                </div>
            </div>
        );
    }
}