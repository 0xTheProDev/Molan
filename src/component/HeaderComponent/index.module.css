.header-content {
    background-color: #FFF !important;
    height: 65px !important;
}

.brand {
    padding-left: 40px !important;
    margin-top: -20px !important;
    user-select: none !important;
    pointer-events: none !important;
    -webkit-user-drag: none !important;
}
