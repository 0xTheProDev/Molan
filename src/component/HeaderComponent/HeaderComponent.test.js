import React from 'react';

import { HeaderComponent } from './HeaderComponent';

describe('Header Component', () => {

  it('should render', () => {
    const wrapper = shallow(<HeaderComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
