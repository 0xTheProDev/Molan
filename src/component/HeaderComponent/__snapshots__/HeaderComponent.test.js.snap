// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render 1`] = `
<div
  className="header-content"
>
  <div
    className="brand"
  >
    <Image
      as="img"
      size="small"
      src="molan.png"
      ui={true}
    />
  </div>
</div>
`;
